File: 프로그래머스/unrated/140108. 문자열 나누기/문자열 나누기.py
def solution(s):    
    answer = 0
    
    tmp1, tmp2 = 0, 0
    x = ''
    for i in range(len(s)):
        if i == len(s) - 1:
            answer += 1
            break
        
        if not x:
            tmp1 += 1
            x = s[i]                
        else:
            if s[i] == x:
                tmp1 += 1
            else:
                tmp2 += 1
        
        if tmp1 == tmp2:
            answer += 1
            tmp1, tmp2 = 0, 0
            x = ''
   
    return answer

File: 프로그래머스/1/140108. 문자열 나누기/문자열 나누기.js
function solution(s) {
    var answer = 0;
    s = s.split('');
        
    while (true) {
        const firstString = s[0];
        
        let cnt1 = 0, cnt2 = 0;
        for (let i = 0; i < s.length; i++) {
            if (s[i] === firstString) cnt1++;
            else cnt2++;     
            if (cnt1 === cnt2) {
                s = s.slice(i + 1, s.length);
                answer++;
                break;
            }
        }
        
        if (s.length === 0) break;
        
        if (cnt1 !== cnt2) {
            answer++;
            break;
        }        
    }    
    
    return answer;
}